File: stacks+queues/Queue.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

class Queue:
    def __init__(self):
        self.first = None
        self.last = None
        self.length = 0
    
    def peek(self):
        return self.first.value
    
    def enqueue(self, value):
        newNode = Node(value)
        if self.length == 0:
            self.first = newNode
            self.last = newNode
        else:
            self.last.next = newNode
            self.last = newNode
        self.length += 1
        return self
    
    def dequeue(self):
        if not self.first:
            return None
        if (self.first == self.last):
            self.last = None
        removed = self.first
        self.first = self.first.next
        self.length -= 1
        return removed.value
    
myQueue = Queue()
myQueue.enqueue(2)
print(myQueue.peek())
myQueue.enqueue(7)
myQueue.enqueue(14)
myQueue.dequeue()
print(myQueue.peek())

File: stacks+queues/Queue.js
class Node {
    constructor(value) {
      this.value = value;
      this.next = null;
    }
  }

  class Queue {
    constructor() {
      this.first = null;
      this.last = null;
      this.length = 0;
    }
    peek() {
        return this.first.value
    }
    enqueue(value) {
        const newNode = new Node(value);
        if (this.length == 0) {
            this.first = newNode;
            this.last = newNode;
        }
        else {
            this.last.next = newNode
            this.last = newNode
        }
        this.length++
        return this
    }
    dequeue() {
        if (!this.first) {
            return None
        }
        if (this.first == this.last) {
            this.last = null
        }
        let removed = this.first
        this.first = this.first.next
        this.length--;
        return removed
    }
    //isEmpty;
  }

  const myQueue = new Queue();
myQueue.enqueue("Joey")
myQueue.enqueue("Marsha")
myQueue.enqueue("Lily")

console.log(myQueue.peek())

myQueue.dequeue();
console.log(myQueue.peek())

  //Joy
  //Matt
  //Pavel
  //Samir